src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+export enum AriaLivePoliteness {
+  off = 'off',
+  polite = 'polite',
+  assertive = 'assertive',
+}
+
+/**
+ * Time in milliseconds before inserting the content into the container
+ */
+export const ARIA_LIVE_TICK: number = 100;
+
+/**
+ * This service handle `aria-live` accessibility attribute. The issue is that you need
+ * to have the DOM Element with attribute `aria-live` before you could insert content
+ * and SR (Screen Reader) pick the change and announce it.
+ *
+ * @remark
+ * This is a private service, nothing here is part of Clarity's public API. It could change at any point in time.
+ *
+ * ```typescript
+ * import { AriaLiveService } from 'src/clr-angular/utils/a11y/aria-live.service';
+ *
+ * @Component({
+ * selector: 'clr-demo-component',
+ * providers: [AriaLiveService],
+ * template: `
+ *   <ng-content></ng-content>
+ * `,
+ * })
+ * export class DemoComponent {
+ *  constructor(ariaLiveService: AriaLiveService) {}
+ *
+ *  public actionThatWillTriggerChange() {
+ *    this.ariaLiveService.announce('message that I want to announce to SR');
+ *  }
+ * }
+ * ```
+ *
+ */
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private ariaLiveElement: HTMLElement;
+  private document: Document;
+  private previousTimeout: ReturnType<typeof setTimeout>;
+
+  constructor(
+    private ngZone: NgZone,
+    @Inject(DOCUMENT) _document: any,
+    @Inject(PLATFORM_ID) private platformId: Object
+  ) {
+    this.document = _document;
+  }
+
+  private id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+  /**
+   * get access to the internal HTML `id` that gonna be used for the AriaLive container.
+   * @return ID of the DOM Element as string.
+   */
+  public getId(): string {
```

Comment:
Why is this still a method? This method doesn't do anything. It should be just a getter property. 
```
get id(): string {
   return this._id;
}
```